medusa/indexers/tmdb/tmdb.py

```diff
@@ -182,11 +182,27 @@ def search(self, series):
         """
         log.debug('Searching for show: {0}', series)
 
-        results = self._show_search(series, request_language=self.config['language'])
-        if not results:
+        results = None
+        # If search term is digit's only, store it and attempt to search by id.
+        show_by_id = None
+
+        try:
+            if series.isdigit():
+                show_by_id = self._get_show_by_id(series, request_language=self.config['language'])
+            results = self._show_search(series, request_language=self.config['language'])
```

Comment:
I think we should rename `_show_search` to `_get_show_by_name` at this point.

Reply:
Sure I can do some refactoring

Reply:
Second thought. Get_show_by_name is different from a search. The name implies it will return one show. I think its more clear to keep the term "search" In it